src/iexamine.cpp

```diff
@@ -56,225 +56,278 @@ void iexamine::gaspump(player *p, map *m, int examx, int examy)
     add_msg(m_info, _("Out of order."));
 }
 
-void iexamine::atm(player *p, map *m, int examx, int examy)
-{
-    (void)m; //unused
-    (void)examx; //unused
-    (void)examy; //unused
-    int choice = -1;
-    const int purchase_cash_card = 1;
-    const int deposit_money = 2;
-    const int withdraw_money = 3;
-    const int transfer_money = 4;
-    const int transfer_all_money = 5;
-    const int cancel = 0;
-    long amount = 0;
-    long max = 0;
-    std::string popupmsg;
-    int pos;
-    int pos2;
-    item *dep;
-    item *with;
+namespace {
+//--------------------------------------------------------------------------------------------------
+//! Implements iexamine::atm(...)
+//--------------------------------------------------------------------------------------------------
+class atm_menu {
+public:
+    // menu choices
+    enum : int {
+        cancel, purchase_card, deposit_money, withdraw_money, transfer_money, transfer_all_money
+    };
+
+    atm_menu()                           = delete;
+    atm_menu(atm_menu const&)            = delete;
+    atm_menu(atm_menu&&)                 = delete;
+    atm_menu& operator=(atm_menu const&) = delete;
+    atm_menu& operator=(atm_menu&&)      = delete;
+
+    explicit atm_menu(player &p) : u {p} {
```

Comment:
Error'd in MinGW32: apparently needs to be const, not grab the player, or not from this sort of list:
C:\Documents and Settings\---\My Documents\GitHub\Cataclysm-DDA\src\iexamine.cpp|76|error: invalid initialization of non-const reference of type 'player&' from an rvalue of type '<brace-enclosed initializer list>'|